modules/gdscript/gdscript_parser.cpp

```diff
@@ -2918,8 +2917,29 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_call(ExpressionNode *p_pre
 				}
 				make_completion_context(COMPLETION_ATTRIBUTE_METHOD, call->callee);
 			} else {
-				// TODO: The analyzer can see if this is actually a Callable and give better error message.
-				push_error(R"*(Cannot call on an expression. Use ".call()" if it's a Callable.)*");
+				if (attribute->index->type == Node::IDENTIFIER) {
+					IdentifierNode *func_name = static_cast<IdentifierNode *>(attribute->base);
+					if (GDScriptParser::get_builtin_type(func_name->name) < Variant::VARIANT_MAX) {
+						IdentifierNode *index = static_cast<IdentifierNode *>(attribute->index);
+						if (index->source == IdentifierNode::UNDEFINED_SOURCE) {
+							// This cast is safe because `typed` should only be set if base is an identifier
+							call->callee = func_name;
+							call->function_name = func_name->name;
+							TypeNode *type = alloc_node<TypeNode>();
+							type->type_chain.push_back(index);
+							reset_extents(type, index);
+							complete_extents(type);
+							call->typed = type;
+						} else {
+							push_error(R"*(Typed constructors can only be called with a valid Type.)*", index);
+						}
+					} else {
+						push_error(R"*(Cannot call on an expression. Use ".call()" if it's a Callable.)*");
+					}
```

Comment:
I feel like the parser is doing too much here. Checking it the type is valid type is the job of the analyzer. A type can be more than identifier, it might be an attribute access (`SomeClass.InnerClass`) or technically even a `preload`ed script.

The only thing the parser should check is if the base is `Array` since it's the only type that has this kind of constructor (can be changed later in the future). At most, check if it's a builtin type and give the error about typed constructors if it's builtin but not an array and if it's not builtin give the error about calling on an expression.